rct229/rulesets/ashrae9012019/section19/section19rule15.py

```diff
@@ -0,0 +1,213 @@
+    HVAC_SYS,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_baseline_system_types import (
+    get_baseline_system_types,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_dict_of_zones_and_terminal_units_served_by_hvac_sys import (
+    get_dict_of_zones_and_terminal_units_served_by_hvac_sys,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_fan_system_object_supply_return_exhaust_relief_total_power_flow import (
+    get_fan_system_object_supply_return_exhaust_relief_total_power_flow,
+)
+from rct229.schema.config import ureg
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all, find_one
+from rct229.utils.pint_utils import ZERO
+from rct229.utils.std_comparisons import std_equal
+
+APPLICABLE_SYS_TYPES = [
+    HVAC_SYS.SYS_9,
+    HVAC_SYS.SYS_10,
+]
+
+REQ_DESIGN_SUPPLY_AIR_TEMP_SETPOINT = 105.0 * ureg("degF")
+
+
+class Section19Rule15(RuleDefinitionListIndexedBase):
+    """Rule 17 of ASHRAE 90.1-2019 Appendix G Section 19 (HVAC - General)"""
+
+    def __init__(self):
+        super(Section19Rule15, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section19Rule15.HVACRule(),
+            index_rmr="baseline",
+            id="19-15",
+            description="For baseline system types 9 & 10, the system design supply airflow rates shall be based on the temperature difference between a supply air temperature set point of 105°F and the design space-heating temperature set point, the minimum outdoor airflow rate, or the airflow rate required to comply with applicable codes or accreditation standards, whichever is greater.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1.2.8.2",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+
+        hvac_info_dict_b = {}
+        for hvac_b in find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+            rmi_b,
+        ):
+            hvac_id_b = hvac_b["id"]
+            hvac_info_dict_b[hvac_id_b] = {}
+            hvac_info_dict_b[hvac_id_b]["all_design_setpoints_105"] = True
+            hvac_info_dict_b[hvac_id_b]["proposed_supply_cfm"] = ZERO.FLOW
+            hvac_info_dict_b[hvac_id_b][
+                "are_all_hvac_sys_fan_objs_autosized"
+            ] = are_all_hvac_sys_fan_objs_autosized(rmi_b, hvac_id_b)
+
+            zones_and_terminal_units_served_by_hvac_sys_dict = (
+                get_dict_of_zones_and_terminal_units_served_by_hvac_sys(hvac_b)
+            )
```

Comment:
The input to this function is `rmi_b`
This function call shall be moved out of the for loop.

Reply:
good point! Addressed.